modules/openapi-generator/src/main/resources/C-libcurl/api-body.mustache

```diff
@@ -11,6 +11,95 @@
     snprintf(dst, 256, "%ld", (long int)(src));\
 }while(0)
 
+{{#operations}}
+{{#operation}}
+{{#allParams}}
+
+{{#isEnum}}
+// Functions for enum {{enumName}} for {{{classname}}}_{{{operationId}}}
+
+
+char* {{enumName}}_ToString({{baseName}}_e {{enumName}}){
+char *{{enumName}}Array[] =  { {{#allowableValues}}{{#enumVars}}"{{{value}}}"{{^-last}},{{/-last}}{{/enumVars}}{{/allowableValues}} };
```

Comment:
Here, the first value when the code is generated will be the actual enum value and its respective number will be 0, so if the enum is mandatory or optional in both cases the enum will be sent. But to avoid this we can make the array start with { NULL , and rest of values}. This will make NULL as 0, so when the enum is optional you can actually skip sending the enum value.

Reply:
ok!

Reply:
Same for model-body.mustache?

Reply:
yes, same for model-body and model-header. If you can push that also it would be great.

Reply:
NULL is a token and cannot be used. I am worried about using NOTHING or UNDEFINED because they could be also values in the enum.
How do you suggest to call the first "NULL" element?

Reply:
I could just decorate "NULL" and it could work.
On the other hand, the String corresponding to the NULL enum would be better be "", since a lot of user code does strcmp etc. Do you agree?

Reply:
These are my opinions and things which I have learned while using the code generator.

you could do something like this: 
```typedef enum  { {{projectName}}_{{classVarName}}_NULL = 0,{{#enumVars}} {{projectName}}_{{classVarName}}_{{{value}}} {{^-last}},{{/-last}}{{/enumVars}} } {{classname}}_e;```

In case of keywords, you have to escape them. So how it will be is like in ToString and FromString, the actual values will be present as these are sent to the destination, and these values are in string format so they will not create problems.  and in model-header, where the enum is defined, it will contain escaped/prefixed values then it will not conflict with C keywords or duplicate values.

Here is an model-body code snippet
```
char* {{classname}}_ToString({{classname}}_e {{classname}}){
char *{{classname}}Array[] =  { "NULL", {{#allowableValues}}{{#enumVars}}"{{{value}}}"{{^-last}},{{/-last}}{{/enumVars}}{{/allowableValues}} };
    return {{classname}}Array[{{classname}}];
}

{{classname}}_e {{classname}}_FromString(char* {{classname}}){
    int stringToReturn = 0;
    char *{{classname}}Array[] =  {"NULL",  {{#allowableValues}}{{#enumVars}}"{{{value}}}"{{^-last}},{{/-last}}{{/enumVars}}{{/allowableValues}} };
    size_t sizeofArray = sizeof({{classname}}Array) / sizeof({{classname}}Array[0]);
    while(stringToReturn < sizeofArray) {
        if(strcmp({{classname}}, {{classname}}Array[stringToReturn]) == 0) {
            return stringToReturn;
        }
        stringToReturn++;
    }
    return 0;
}
```

model-header snippet:
```
typedef enum  { {{projectName}}_{{classVarName}}_NULL = 0,{{#enumVars}} {{projectName}}_{{classVarName}}_{{{value}}} {{^-last}},{{/-last}}{{/enumVars}} } {{classname}}_e;
```

There is a reason I am using ```{{projectName}}_{{classVarName}}_{{{value}}}```, the codebase I have contains a lot of repeated enum values definitions and C keywords, to overcome all these issues I have prefixed everything thing(functions, enums, etc) with these prefixes.


Also for your second comment, string compare will not create problems as enums in the code works as int values and if there is strcmp used anywhere in the code except FromString function then it is not good.

Reply:
The second comment was about using NULL, "" or "NULL", and in fact with "NULL" everything is fine.

The approach
`typedef enum  { {{projectName}}_{{classVarName}}_NULL = 0,{{#enumVars}} {{projectName}}_{{classVarName}}_{{{value}}} {{^-last}},{{/-last}}{{/enumVars}} } {{classname}}_e;`
it does not work in the use case where the values are repeated among two enums used in the same model, e.g.:
```
  schemas:
    '200':
      title: Successful response
      type: object
      properties:
        weather:
          type: string
          description: (more info Weather condition codes)
        secure:
          type: string
          enum:
            - NOT_SECURE
            - CERTIFICATE
            - TOKEN
        secure2:
          type: string
          enum:
            - NOT_SECURE
            - CERTIFICATE
            - TOKEN
```
Thus, we need at least {{enumName}} in the decoration.
Then, if there is repetition between different models, and we need also {{classVarName}}.
If we put there also `{{projectName}}`, an example of enum name is now `openweathermap_api__200_SECURE_NOT_SECURE`. Isn't it a little heavy? Would it be better without {{projectName}}?

Reply:
It should be ok now. Anyway, I have not covered classes enums. I will work on that on a future PR, after somebody provides YAMLs to play against.

Reply:
@zhemant Is it ok the way I changed the code?

Reply:
@michelealbano Yes it is okay. Sorry for late replies